### PR TITLE
Evict cache every 12 hours

### PR DESCRIPTION
**Pull request by southeo**
CacheManager = holds names of caches (pidname, qid)
CacheEvictionService = every 12 hours, evicts caches in cache manager

Will need to update deployment file with eviction frequency (43200000 ms)

### CODE REVIEW

File: src/main/java/eu/dissco/core/handlemanager/service/CacheEvictionService.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class CacheEvictionService {
+
+  private final CacheManager cacheManager;
+
+  @Scheduled(fixedRateString = "${caching.spring.ttl}")
```
Comment (southeo) [inline src/main/java/eu/dissco/core/handlemanager/service/CacheEvictionService.java:17]: will need to update deployment file...

File: src/main/java/eu/dissco/core/handlemanager/service/CacheEvictionService.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  public void evictAllCaches() {
+    log.info("Evicting all caches");
+    cacheManager.getCacheNames()
```
Comment (samleeflang) [inline src/main/java/eu/dissco/core/handlemanager/service/CacheEvictionService.java:20]: I thought there would be an easier way to do this, but look like you need the scheduler.
Also saw this example
https://www.baeldung.com/spring-setting-ttl-value-cache

**Review by samleeflang — APPROVED** [review]
We don't need this for ROR as well?

**Comment by southeo** [comment]
> We don't need this for ROR as well?

We use the same method to resolve PID types and rors: 

```
@Cacheable("pidName")
public String getObjectName(String pid) {...}
```

Both RORs and FDO types have the field "name" in the same place! Lucky for us. 

https://api.ror.org/organizations/0566bfb96
https://hdl.handle.net/21.T11148/d8de0819e144e4096645

File: src/main/java/eu/dissco/core/handlemanager/service/CacheEvictionService.java (hunk 2 above)
Comment (southeo) [reply src/main/java/eu/dissco/core/handlemanager/service/CacheEvictionService.java:20]: Yeah that's what I used to write this. Is there something from this example you're not seeing?

**Pull request merged**